src/plugins/vis_type_timeseries/server/plugin.ts

```diff
@@ -91,16 +93,34 @@ export class VisTypeTimeseriesPlugin implements Plugin<VisTypeTimeseriesSetup> {
       logger,
       router,
       searchStrategyRegistry,
+      getEsShardTimeout: () =>
+        globalConfig$
+          .pipe(
+            first(),
+            map((config) => config.elasticsearch.shardTimeout.asMilliseconds())
+          )
+          .toPromise(),
+      getIndexPatternsService: async (requestContext) => {
+        const [, { data }] = await core.getStartServices();
+
+        return await data.indexPatterns.indexPatternsServiceFactory(
+          requestContext.core.savedObjects.client,
+          requestContext.core.elasticsearch.client.asCurrentUser
+        );
+      },
     };
 
+    searchStrategyRegistry.addStrategy(new DefaultSearchStrategy(framework));
+    searchStrategyRegistry.addStrategy(new RollupSearchStrategy(framework));
+
     visDataRoutes(router, framework);
```

Comment:
two similar cases but we execute it with different arguments. I think if we pass `router` as param.  In this case `router` can be also removed from `Framework` object.  + we can remove `VisTypeTimeseriesRouter`
```
visDataRoutes(router, framework);
fieldsRoutes(framework);
```

Reply:
make sense, done